sklearn/tree/_splitter.pyx

```diff
@@ -416,7 +416,7 @@ cdef class BestSplitter(BaseDenseSplitter):
                     sort(Xf + start, samples + start, end - start)
 
                 if Xf[end - 1] <= Xf[start] + FEATURE_THRESHOLD:
-                    features[f_j], features[n_total_constants] = features[n_total_constants], current.feature
+                    features[f_j], features[n_total_constants] = features[n_total_constants], features[f_j]
```

Comment:
is this correct?

Reply:
yes, it is, check here. https://github.com/scikit-learn/scikit-learn/pull/14445/files#diff-23467bb90c859a3e381a82626abe64d6L396

Reply:
I guess its readability trumps the reference lookup cost then.